src/solvers/dgsem_tree/dg_1d.jl

```diff
@@ -287,50 +289,88 @@ function calc_volume_integral!(du, u,
 end
 
 
-
 @inline function fv_kernel!(du, u,
                             mesh::Union{TreeMesh{1}, StructuredMesh{1}},
                             nonconservative_terms, equations,
                             volume_flux_fv, dg::DGSEM, cache, element, alpha=true)
-  @unpack fstar1_threaded = cache
+  @unpack fstar1_L_threaded, fstar1_R_threaded = cache
   @unpack inverse_weights = dg.basis
 
   # Calculate FV two-point fluxes
-  fstar1 = fstar1_threaded[Threads.threadid()]
-  calcflux_fv!(fstar1, u, mesh, nonconservative_terms, equations, volume_flux_fv,
+  fstar1_L = fstar1_L_threaded[Threads.threadid()]
+  fstar1_R = fstar1_R_threaded[Threads.threadid()]
+  calcflux_fv!(fstar1_L, fstar1_R, u, mesh, nonconservative_terms, equations, volume_flux_fv,
                dg, element, cache)
 
   # Calculate FV volume integral contribution
   for i in eachnode(dg)
     for v in eachvariable(equations)
       du[v, i, element] += ( alpha *
-                             (inverse_weights[i] * (fstar1[v, i+1] - fstar1[v, i])) )
+                             (inverse_weights[i] * (fstar1_L[v, i+1] - fstar1_R[v, i])) )
 
     end
   end
 
   return nothing
 end
 
-@inline function calcflux_fv!(fstar1, u::AbstractArray{<:Any,3},
+
+@inline function calcflux_fv!(fstar1_L, fstar1_R, u::AbstractArray{<:Any,3},
                               mesh::Union{TreeMesh{1}, StructuredMesh{1}},
                               nonconservative_terms::Val{false},
                               equations, volume_flux_fv, dg::DGSEM, element, cache)
 
-  fstar1[:, 1,           ] .= zero(eltype(fstar1))
-  fstar1[:, nnodes(dg)+1,] .= zero(eltype(fstar1))
+  fstar1_L[:, 1           ] .= zero(eltype(fstar1_L))
+  fstar1_L[:, nnodes(dg)+1] .= zero(eltype(fstar1_L))
+  fstar1_R[:, 1           ] .= zero(eltype(fstar1_R))
+  fstar1_R[:, nnodes(dg)+1] .= zero(eltype(fstar1_R))
 
   for i in 2:nnodes(dg)
     u_ll = get_node_vars(u, equations, dg, i-1, element)
-    u_rr = get_node_vars(u, equations, dg, i,   element)
+    u_rr = get_node_vars(u, equations, dg, i  , element)
     flux = volume_flux_fv(u_ll, u_rr, 1, equations) # orientation 1: x direction
-    set_node_vars!(fstar1, flux, equations, dg, i)
+    set_node_vars!(fstar1_L, flux, equations, dg, i)
+    set_node_vars!(fstar1_R, flux, equations, dg, i)
   end
 
   return nothing
 end
 
 
+@inline function calcflux_fv!(fstar1_L, fstar1_R, u::AbstractArray{<:Any,3},
+                              mesh::TreeMesh{1}, # TODO: Should we add StructuredMesh{1} as well?
```

Comment:
Okay, I can add a source terms and well balanced test for `StructuredMesh` in 1D and 2D. We get it for free basically.

Reply:
Nevermind, the StructuredMesh in 1D cannot handle nonconservative terms. So the `interface_flux` and `boundary_flux` would need to be altered to incorporate this capability. Since we do not have a use case needing this (especially because StructuredMesh and TreeMesh seem equivalent to me in 1D, correct me if I am wrong @sloede)I would prefer to not bother. However, `StructuredMesh2D` is already implemented to handle noncons. So I added a well-balancedness test for that mesh.

Reply:
Okay, thanks!